src/box/tuple_compare.cc

```diff
@@ -232,34 +232,53 @@ mp_compare_decimal(const char *lhs, const char *rhs)
 
 }
 
+static inline int
+compare_decimal_double_infinite(double rhs)
```

Comment:
The function also handles `nan`. "Infinity" is "not a number", but "not a number" is not "infinity". It means it might be better to use `nan` in the name instead of `infinite`.

Reply:
Agree.